sdk/communication/communication-phone-numbers/src/generated/src/phoneNumbersClient.ts

```diff
@@ -19,8 +20,8 @@ export class PhoneNumbersClient extends PhoneNumbersClientContext {
    */
   constructor(endpoint: string, options?: PhoneNumbersClientOptionalParams) {
     super(endpoint, options);
-    this.phoneNumbers = new PhoneNumbers(this);
+    this.phoneNumbers = new PhoneNumbersImpl(this);
   }
 
-  phoneNumbers: PhoneNumbers;
+  phoneNumbers: PhoneNumbersImpl;
```

Comment:
Here I had to manually change the type of `phoneNumbers` to be `PhoneNumbersImpl`. Before it was just `PhoneNumbers` and that's an interface, which created an error in [this line](https://github.com/Azure/azure-sdk-for-js/blob/20b7ba04f6b1353496c8ae73cd644e2bf6716714/sdk/communication/communication-phone-numbers/src/phoneNumbersClient.ts#L115):
https://github.com/Azure/azure-sdk-for-js/blob/20b7ba04f6b1353496c8ae73cd644e2bf6716714/sdk/communication/communication-phone-numbers/src/phoneNumbersClient.ts#L115

This is the error:
```
Type 'PhoneNumbers' is missing the following properties from type 'PhoneNumbersImpl': client, listPhoneNumbersPagingPage, listPhoneNumbersPagingAll, _listPhoneNumbers, and 2 more.
```

@sarangan12 do you know if this could be a bug with the code generator? or if the expected output is indeed for `phoneNumbers` to be of type `PhoneNumbers`?

Reply:
@JonathanCrd The `PhoneNumbersImpl` is a class that is pointing to the implementation of the **operation** that is defined [here](https://github.com/Azure/azure-sdk-for-js/blob/18b4f9bb81b335063a45e491eb9ad272f994d3ab/sdk/communication/communication-phone-numbers/src/generated/src/operations/phoneNumbers.ts). It is not a client.

So, In line 56, when you say such as:

```
/**
   * A reference to the auto-generated PhoneNumber HTTP client.
   */
  private readonly client: GeneratedClient;
```

You should be pointing to the client `PhoneNumbersClient` defined [here](https://github.com/Azure/azure-sdk-for-js/blob/18b4f9bb81b335063a45e491eb9ad272f994d3ab/sdk/communication/communication-phone-numbers/src/generated/src/phoneNumbersClient.ts)

And in the places where you are calling, it should be called such as `this.client.phoneNumber.list...(`. Now, if you want to be concise then in your line 56, change the logic to 

```
/**
   * A reference to the phoneNumbers operation group.
   */
  private readonly phoneNumbers: PhoneNumbers;
```

and then in line 115, you have to change it as:

```
this.phoneNumbers  = new PhoneNumbersGeneratedClient(url, pipeline).phoneNumbers
```

Though I would not suggest to this approach. The correct approach is the first method. The Operation Implementation should not be referenced in the custom layer. Let me know if you still have any more questions.

Reply:
Thank you so much, It's clearer for me now.  Already made the suggested changes 👍🏼